### PR TITLE
[Fabric] Move E2E fabric app to WinAppSDK and lifted composition

### PR DESCRIPTION
**Pull request by acoates-ms**
## Description
- Rewrite E2E fabric test app using WinAppSDK and lifted composition

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12242)

### CODE REVIEW

File: packages/e2e-test-app-fabric/jest.config.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ const defaultTransform = [
   require('@rnw-scripts/babel-node-config'),
 ];
 
+const appWorkingDir = 'windows\\x64\\Release\\RNTesterApp-Fabric';
```
Comment (chiaramooney) [inline packages/e2e-test-app-fabric/jest.config.js:19]: I think this path should be 'windows\\x64\\Release\\' instead. I believe that should fix the issue.

File: packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp [hunk 2]
```diff
@@ -51,263 +46,200 @@ struct DeviceInfo {
...
-      CW_USEDEFAULT,
-      nullptr,
-      nullptr,
-      WindowData::s_instance,
-      windowData.get());
-
-  WINRT_VERIFY(hwnd);
+  // Create a top-level window.
+  auto window = winrt::Microsoft::UI::Windowing::AppWindow::Create();
+  window.Title(appName);
```
Comment (jonthysell) [inline packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp:147]: Should we expose a turbomodule so fabric apps can change their title from JS?

**Review by jonthysell — APPROVED** [review]

**Comment by jonthysell** [comment]
Is there an issue/tracking updating the new app template to use WinAppSDK?

File: packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp (hunk 2 above)
Comment (acoates-ms) [reply packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp:147]: That assumes there is one window.  -- And we know what it is, and some other things.  I think that would have to be part of a larger story to work out what windowing from JS looks like.

**Pull request merged**